### PR TITLE
return 404 for bad id certificate

### LINKED ISSUE #432: 500 on bad certificate number
# Problem

When I try and add a bad certificate number, then I get a 500, I should get a 404.
See:
http://staging.changelog.qgis.org/en/qgis/certificate/0246242/

# Proposed Solution

Return a 404

### PR DESCRIPTION
**Pull request by ann26**
fix #432 
peek 2017-06-12 17-31

### CODE REVIEW

File: django_project/certification/views/certificate.py [hunk 1]
```diff
@@ -125,7 +125,8 @@ def get_object(self, queryset=None):
         if queryset is None:
             queryset = self.get_queryset()
             certificateID = self.kwargs.get('id', None)
-            if certificateID:
+            certificate = Certificate.objects.all().values_list('certificateID', flat=True)
+            if certificateID in certificate:
```
Comment (cchristelis) [inline django_project/certification/views/certificate.py]: why not rather use the `get or 404` pattern here? I think we should rather use the DB to search for a record than return all the objects and search in python...

**Comment by timlinux** [comment]
Maybe rather than returning a 404 it would be better to show a big red 'X' and a note underneath saying that no certificate matching this number has been issued.

File: django_project/certification/views/certificate.py (hunk 1 above)
Comment (ann26) [inline django_project/certification/views/certificate.py]: ok, I've fixed it

**Comment by ann26** [comment]
@timlinux ok, I fixed it
peek 2017-06-14 11-53

**Comment by timlinux** [comment]
Perfect, thanks!

**Pull request merged**